drgn_tools/iscsi.py

```diff
@@ -62,8 +143,95 @@ def for_each_iscsi_session(prog: Program) -> Iterator[Object]:
             address=cls_session.dd_data,
         )
 
+def get_iscsi_info(iscsi_conn: Object, prog: Program ) -> Dict[str, Any]:
 
-def print_iscsi_sessions(prog: Program) -> None:
+    iscsi_session = iscsi_conn.session
+    iscsi_cls_conn = iscsi_conn.cls_conn
+    iscsi_cls_session = iscsi_session.cls_session
+    iscsi_tcp_conn = cast("struct iscsi_tcp_conn *", iscsi_conn.dd_data)
+    iscsi_sw_tcp_conn = cast("struct iscsi_sw_tcp_conn *", iscsi_tcp_conn.dd_data)
+    iscsi_socket = iscsi_sw_tcp_conn.sock
+    iscsi_sock = iscsi_socket.sk
+    scsi_host = iscsi_session.host
+    iscsi_host_data = scsi_host.hostdata
+    iscsi_host = cast("struct iscsi_host *", iscsi_host_data)
+
+    try:
+        if iscsi_sock.__sk_common.skc_family == 2:
+            current_address = iscsi_sock.__sk_common.skc_daddr
+            current_address = prog.read_u32(current_address.address_of_())
+            current_ip = socket.inet_ntoa(bytes([(current_address) & 0xFF,
+                                        (current_address >> 8) & 0xFF,
+                                        (current_address >> 16) & 0xFF,
+                                        (current_address >> 24) & 0xFF]))
+            current_port = iscsi_sock.__sk_common.skc_dport
+            current_port = prog.read_u16(current_port.address_of_())
+            current_port = socket.ntohs(current_port)
+            local_address = iscsi_sock.__sk_common.skc_rcv_saddr
+            local_address = prog.read_u32(local_address.address_of_())
+            local_ip = socket.inet_ntoa(bytes([(local_address) & 0xFF,
+                                        (local_address >> 8) & 0xFF,
+                                        (local_address >> 16) & 0xFF,
+                                        (local_address >> 24) & 0xFF]))
```

Comment:
For IP addresses, you should use `get_ipv4()` which is defined in `drgn_tools/crash_net.py`, see its use there. The kernel always stores the IP addresses in network byte order, so as long as you use that function, you don't need to do byte swapping.

For the port, only do the byte swap if the program is little endian. (which is true for all the architectures we use, but we really should do it correctly). For example, see these lines in the `crash_net.py` file:
```
            dport = int(skc.skc_dport)
            if prog.platform.flags & PlatformFlags.IS_LITTLE_ENDIAN:
                dport = socketlib.ntohs(dport)
```